app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class Api::V1::UsersController < Api::V1::BaseController
+  before_action :set_user, only: [:show, :update]
+  before_action :authenticate_user!
+  before_action :check_access, only: [:update]
+
+  def index
+    @users = paginate(User.all)
+    @options = {  meta: meta_attributes(@users) }
+    render json: Api::V1::UsersSerializer.new(@users, @options)
+  end
+
+  def show
+    render json: Api::V1::UserSerializer.new(
+      @user, params: { has_email_access: @user.eql?(@current_user) })
+  end
+
+  def update
+    @user.update!(user_params)
+    if @user.update(user_params)
+      render json: Api::V1::UserSerializer.new(
+        @user, params: { has_email_access: @user.eql?(@current_user) }
+      ), status: :accepted
+    else
+      invalid_resource!(@user.errors)
+    end
+  end
```

Comment:
Can you add one more endpoint

/api/v1/me

which returns current user

Reply:
Sure!